### PR TITLE
Fix `check_config_attributes`: check all configuration classes

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

As @NielsRogge points out to me, the `check_config_attributes` doesn't check all configuration classes (to make sure all `__init__` arguments are really used).

This is because `CONFIG_MAPPING` doesn't contain all configuration classes, in particular, the vision/text config classes for models like Blip2 or CLIP.

This PR fixes this issue, and remove some unused arguments/attributes in some configuration classes.

### CODE REVIEW

File: src/transformers/models/blip/configuration_blip.py
Comment (ydshieh) [inline src/transformers/models/blip/configuration_blip.py:1]: @younesbelkada Could you check the changes in this file? Thank you.

File: src/transformers/models/blip_2/configuration_blip_2.py
Comment (ydshieh) [inline src/transformers/models/blip_2/configuration_blip_2.py:1]: @NielsRogge Could you check the changes in this file? Thank you.

File: src/transformers/models/clap/configuration_clap.py
Comment (ydshieh) [inline src/transformers/models/clap/configuration_clap.py:1]: @ArthurZucker Could you check the changes in this file? Thank you.

File: src/transformers/models/pix2struct/configuration_pix2struct.py
Comment (ydshieh) [inline src/transformers/models/pix2struct/configuration_pix2struct.py:1]: @younesbelkada Could you check the changes in this file? Thank you.

File: src/transformers/models/sam/configuration_sam.py
Comment (ydshieh) [inline src/transformers/models/sam/configuration_sam.py:1]: @ArthurZucker  Could you check the changes in this file? Thank you.

File: src/transformers/models/speecht5/configuration_speecht5.py
Comment (ydshieh) [inline src/transformers/models/speecht5/configuration_speecht5.py:1]: @hollance  Could you check the changes in this file? Thank you.

**Review by younesbelkada — APPROVED** [review]
Thank you! All good on BLIP & Pix2Struct side

File: src/transformers/models/speecht5/configuration_speecht5.py
Comment (hollance) [reply src/transformers/models/speecht5/configuration_speecht5.py:1]: It's not used explicitly in the modeling code, but I'd argue it's still important to know what the sampling rate will be of the output audio. The config is really the only place to put it.

Comment (ydshieh) [reply src/transformers/models/speecht5/configuration_speecht5.py:1]: Agree. I think same for `num_channels`. I will ask on a slack channel.

**Review by amyeroberts — APPROVED** [review]
Really nice clean up :) 

Happy to merge once change in config vars have been checked by the others

**Review by ArthurZucker — COMMENTED** [review]
It looks good to me for sam and clap ! thanks for the cleanup

**Pull request merged**